### PR TITLE
JBPM-4646 - Case management - futher improvements for queries and cas…

### PR DESCRIPTION
**Pull request by mswiderski**
…e id generator to make it pluggable in kie server

### CODE REVIEW

**Comment by mswiderski** [comment]
jenkins retest this

File: jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/CaseRuntimeDataServiceImpl.java [hunk 1]
```diff
@@ -317,7 +369,7 @@ public CaseDefinition getCase(String deploymentId, String caseDefinitionId) {
             .forEach(cmi -> milestones.add(cmi));
         }
         
-        return milestones;
+        return applyPaginition(milestones, queryContext);
```
Comment (ibek) [inline jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/CaseRuntimeDataServiceImpl.java]: typo - applyPagination

Comment (mswiderski) [inline jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/CaseRuntimeDataServiceImpl.java]: fixed

**Comment by ibek** [comment]
+1 for merge

**Pull request merged**